src/sys/unix/waker.rs

```diff
@@ -222,7 +224,10 @@ mod pipe {
     impl WakerInternal {
         pub fn new() -> io::Result<WakerInternal> {
             let mut fds = [-1; 2];
+            #[cfg(not(target_os = "aix"))]
             syscall!(pipe2(fds.as_mut_ptr(), libc::O_NONBLOCK | libc::O_CLOEXEC))?;
+            #[cfg(target_os = "aix")]
+            syscall!(pipe(fds.as_mut_ptr()))?;
```

Comment:
You still need to set the `libc::O_NONBLOCK | libc::O_CLOEXEC` otherwise this will not work.